diff --git a/rtl/tlul_adapter_sram.sv b/rtl/tlul_adapter_sram.sv
index 5556f94..e0412b4 100644
--- a/rtl/tlul_adapter_sram.sv
+++ b/rtl/tlul_adapter_sram.sv
@@ -315,7 +315,9 @@ module tlul_adapter_sram
       d_data   : d_data,
       d_user   : '{default: '0, data_intg: data_intg, capability: d_cap},
       d_error  : d_valid && d_error,
+      /* verilator lint_off UNOPTFLAT */
       a_ready  : (gnt_i | error_internal) & reqfifo_wready & sramreqfifo_wready
+      /* verilator lint_on UNOPTFLAT */
   };
 
   // a_ready depends on the FIFO full condition and grant from SRAM (or SRAM arbiter)
