diff --git a/rtl/rv_plic.sv b/hw/top_earlgrey/ip_autogen/rv_plic/rtl/rv_plic.sv
index b9b77c1436..be7dd0b5f2 100644
--- a/rtl/rv_plic.sv
+++ b/rtl/rv_plic.sv
@@ -36,9 +36,7 @@ module rv_plic import rv_plic_reg_pkg::*; #(
 
   // Interrupt notification to targets
   output [NumTarget-1:0] irq_o,
-  output [SRCW-1:0]      irq_id_o [NumTarget],
-
-  output logic [NumTarget-1:0] msip_o
+  output [SRCW-1:0]      irq_id_o [NumTarget]
 );
 
   rv_plic_reg2hw_t reg2hw;
@@ -295,10 +293,8 @@ module rv_plic import rv_plic_reg_pkg::*; #(
   assign complete_id[0] = reg2hw.cc0.q;
   assign hw2reg.cc0.d   = cc_id[0];
 
-  ///////////////////
-  // MSIP register //
-  ///////////////////
-  assign msip_o[0] = reg2hw.msip0.q;
+  logic unused_msip;
+  assign unused_msip = reg2hw.msip0.q;
 
   ////////
   // IP //
@@ -386,7 +382,6 @@ module rv_plic import rv_plic_reg_pkg::*; #(
   `ASSERT_KNOWN(TlDValidKnownO_A, tl_o.d_valid)
   `ASSERT_KNOWN(TlAReadyKnownO_A, tl_o.a_ready)
   `ASSERT_KNOWN(IrqKnownO_A, irq_o)
-  `ASSERT_KNOWN(MsipKnownO_A, msip_o)
   for (genvar k = 0; k < NumTarget; k++) begin : gen_irq_id_known
     `ASSERT_KNOWN(IrqIdKnownO_A, irq_id_o[k])
   end
