diff --git a/vendor/lowrisc_ip/ip/prim/prim.core b/vendor/lowrisc_ip/ip/prim/prim.core
index f348de3..6dc5f67 100644
--- a/vendor/lowrisc_ip/ip/prim/prim.core
+++ b/vendor/lowrisc_ip/ip/prim/prim.core
@@ -21,7 +21,6 @@ filesets:
       - lowrisc:prim:flop_2sync
       - lowrisc:prim:arbiter
       - lowrisc:prim:fifo
-      - lowrisc:prim:alert
       - lowrisc:prim:esc
       - lowrisc:prim:subreg
       - lowrisc:prim:cipher
