diff --git a/rtl/usbdev_reg_top.sv b/rtl/usbdev_reg_top.sv
index d7d9322..2161da8 100644
--- a/rtl/usbdev_reg_top.sv
+++ b/rtl/usbdev_reg_top.sv
@@ -49,8 +49,8 @@ module usbdev_reg_top (
   // outgoing integrity generation
   tlul_pkg::tl_d2h_t tl_o_pre;
   tlul_rsp_intg_gen #(
-    .EnableRspIntgGen(1),
-    .EnableDataIntgGen(1)
+    .EnableRspIntgGen(0),
+    .EnableDataIntgGen(0)
   ) u_rsp_intg_gen (
     .tl_i(tl_o_pre),
     .tl_o(tl_o)
